src/resolvelib/structs.py

```diff
@@ -116,22 +116,23 @@ class _FactoryIterableView(object):
     """
 
     def __init__(self, factory):
-        self._factory = factory
+        self._iterable = factory()
```

Comment:
Is it possible to not call the factory here? It would be nice to not needing to execute the code if the iterable is never accessed.